appcelerator/titanium_mobile
# fix(android): string reference equality in service code

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-27948

**Summary:**
- Android service code was wrongly doing string reference equality instead of value equality.
- Minor issue since it only affects debug logging.
- Changed to satisfy security scanning apps such as Veracode.

**Test:**
1. Add the below "tiapp.xml" setting to your project. (You need to set debug to `true`.)
2. Add the below "app.js" and "MyService.js" to your project.
3. Build with debug logging enabled. Ex: `appc run -p android -l debug`
4. In the log, verify you see a message similar to the below.
`[DEBUG] TiJSService: (main) [##,##] Eval JS Service: MyService.js (app://MyService.js)`

_tiapp.xml_
```xml
<ti:app>
	<property name="ti.android.debug" type="bool">true</property>
	<android xmlns:android="http://schemas.android.com/apk/res/android">
		<services>
			<service url="MyService.js"/>
		</services>
	</android>
</ti:app>
```

_./Resources/app.js_
```javascript
Ti.Android.startService(Ti.Android.createServiceIntent({
	url: "MyService.js",
}));
```

_./Resources/MyService.js_
```javascript
Ti.API.info("@@@ Service started.");
```
